tests/flows/check_upgrade_access.go

```diff
@@ -0,0 +1,86 @@
+package flows
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	"github.com/ava-labs/teleporter/tests/interfaces"
+	"github.com/ava-labs/teleporter/tests/utils"
+	"github.com/ethereum/go-ethereum/crypto"
+	. "github.com/onsi/gomega"
+)
+
+func CheckUpgradeAccess(network interfaces.Network) {
+	subnetInfo := network.GetPrimaryNetworkInfo()
+	_, fundedKey := network.GetFundedAccountInfo()
+
+	//
+	// Deploy ExampleMessenger to Subnets A and B
+	//
+	ctx := context.Background()
+	teleporterAddress := network.GetTeleporterContractAddress()
+	_, exampleMessenger := utils.DeployExampleCrossChainMessenger(
+		ctx, fundedKey, subnetInfo,
+	)
+
+	// Try to call updateMinTeleporterVersion from a non owner account
+	nonOwnerKey, err := crypto.GenerateKey()
+	Expect(err).Should(BeNil())
+	nonOwnerAddress := crypto.PubkeyToAddress(nonOwnerKey.PublicKey)
+
+	// Transfer native assets to the non owner account
+	fundAmount := big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(1)) // 10eth
+	fundNonOwnerTx := utils.CreateNativeTransferTransaction(
+		ctx,
+		subnetInfo,
+		fundedKey,
+		nonOwnerAddress,
+		fundAmount)
+	utils.SendTransactionAndWaitForSuccess(ctx, subnetInfo, fundNonOwnerTx)
+
+	nonOwnerOpts, err := bind.NewKeyedTransactorWithChainID(
+		nonOwnerKey, subnetInfo.EVMChainID)
+	Expect(err).Should(BeNil())
+	ownerOpts, err := bind.NewKeyedTransactorWithChainID(
+		fundedKey, subnetInfo.EVMChainID)
+	Expect(err).Should(BeNil())
+
+	_, err = exampleMessenger.PauseTeleporterAddress(nonOwnerOpts, teleporterAddress)
+	Expect(err).ShouldNot(BeNil())
```

Comment:
I think we actually need to call `WaitForTransactionFailure` after this. ABI binding calls that mutate state don't wait for the transaction to be included in a block AFAIK, so we could end up calling `IsTeleporterAddressPaused()` because this transaction is included, making the check meaningless.

Reply:
since the call gets reverted, the abi bind is implemented to return a `nil` transaction, and would have a nil reference if called with `WaitForTransactionFailure` after. Added the check that the error contains the expected revert message.